From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dinesh Manajipet <saidinesh5@gmail.com>
Date: Fri, 16 Dec 2016 15:49:06 +0530
Subject: [PATCH] (hybris) Override binding to netlink multicast group from env

uevent.c will override the default netlink group "0xffffffff" with the
value from the environment variable UEVENT_NETLINK_GROUPS, if exists.

Useful to prevent vendor blobs like hvdcp, charger_monitor to prevent
reading uevents from systemd-udevd etc..

Change-Id: I7a896df61ae37db6f8d66ef3e9a08569b65245e2
---
 libcutils/uevent.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/libcutils/uevent.cpp b/libcutils/uevent.cpp
index f01a479cd8f58415e7b65ab3cbd406cb3dd10019..c87264f1234cf2d93cae704723a341245a5b9ab3 100644
--- a/libcutils/uevent.cpp
+++ b/libcutils/uevent.cpp
@@ -19,6 +19,7 @@
 #include <errno.h>
 #include <stdint.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <strings.h>
 #include <sys/socket.h>
@@ -144,6 +145,11 @@ int uevent_bind(int socket) {
             .nl_pid = 0,
             .nl_groups = 0xffffffff,
     };
+
+    char* nl_groups_env = getenv("UEVENT_NETLINK_GROUPS");
+    if (nl_groups_env != NULL)
+       addr.nl_groups = strtoul(nl_groups_env, NULL, 0);
+
     return bind(socket, (struct sockaddr*)&addr, sizeof(addr));
 }
 
